warehouse/query-core/src/main/java/datawave/query/discovery/FindLiteralsAndPatternsVisitor.java

```diff
@@ -44,17 +42,17 @@ public static QueryValues find(JexlNode root) {
 
     @Override
     public Object visit(ASTStringLiteral node, Object data) {
-        // strings are always wrapped in a reference node, so the op is the gparent
-        JexlNode op = node.jjtGetParent().jjtGetParent();
+        // strings are NEVER wrapped in a reference node, so the op is the parent
+        JexlNode op = node.jjtGetParent();
         if (op instanceof ASTEQNode) {
-            JexlNode field = JexlNodes.otherChild(op, node.jjtGetParent()).jjtGetChild(0);
+            JexlNode field = JexlNodes.otherChild(op, node);
             if (log.isTraceEnabled()) {
-                log.trace("Found field " + JexlASTHelper.deconstructIdentifier(field.image) + "==" + node.getLiteral());
+                log.trace("Found field " + JexlASTHelper.deconstructIdentifier(((ASTIdentifier) field).getName()) + "==" + node.getLiteral());
```

Comment:
Are we sure this will be an identifier ?

Reply:
In this method we have a string literal with an equals node as a parent.  Technically there has NEVER been a guarantee that the sibling node is an identifier, but that's the assumption this code has always made (as evidenced by the deconstructIdentifier call).  

This brings up an issue with our current understanding of how Jexl works.  In Jexl 2.x every node has an image property (string), but only a small subset of nodes actually use the image variable (String literal, Number literal, Identifier).  The image isn't always accurate either - for a negative number, the number literal will actually have the positive number as the image and will be wrapped in a unary minus node to signify that it's negative.  I mention this because I think we have made some inaccurate assumptions about how images work in our code.  

Getting back to this method, we have been relying on the sibling node to have a valid image value, but that will not be the case unless the sibling is a literal or an identifier.  If the other side of the equals node is a function or a method, then field.image would have returned null.  So, while it seems pretty clear to me that this code expected the sibling to be an identifier, I suppose I can update this to call JexlNodes.getImage instead, which will behave similarly to the previous logic.

Reply:
Updated.